tests/TestCIT.py

```diff
@@ -1,26 +1,41 @@
-from itertools import chain, combinations
+# from itertools import chain, combinations
+import unittest
 
 import numpy as np
+from causallearn.utils.cit import CIT
+# from causallearn.utils.cit import (chisq, chisq_notoptimized, gsq, gsq_notoptimized)
 
-from causallearn.utils.cit import (chisq, chisq_notoptimized, gsq,
-                                   gsq_notoptimized)
 
+class TestCIT(unittest.TestCase):
+    def test_fisherz_singularity_problem(self):
+        X1 = X2 = np.random.normal(size=1000)
+        X = np.array([X1, X2])
 
-def test_new_old_gsq_chisq_equivalent(self):
-    def powerset(iterable):
-        return chain.from_iterable(combinations(list(iterable), r) for r in range(len(iterable) + 1))
+        cit = CIT(data=X, method='fisherz')
 
-    def _unique(column):
-        return np.unique(column, return_inverse=True)[1]
+        try:
+            cit.fisherz(0, 1, tuple())
+        except ValueError:
+            print('Catch Singularity Problem')
+            return
 
-    data_path = "data_discrete_10.txt"
-    data = np.loadtxt(data_path, skiprows=1)
-    data = np.apply_along_axis(_unique, 0, data).astype(np.int32)
-    cardinalities = np.max(data, axis=0) + 1
+        assert False

```

Comment:
This is not a good way to write test in this case.

I did a simple google search using keyword: "python test expect exception", and this result looks reasonable: https://stackoverflow.com/questions/129507/how-do-you-test-that-a-python-function-throws-an-exception

Maybe you can try it. (Or maybe you can even find something better! :) )

Reply:
Thanks for your suggestion! I will try it.